### ISSUE TITLE
Lake serve has become 50% slower since 3.2.1

### ISSUE THREAD

**Issue by gebner** [leanprover/lake#103]
```
cd lake
git checkout v3.2.1
time lake serve </dev/null    # 440ms
git checkout ff619a3397bed8b6474ac12e5cf7f78cf59a34da
time lake serve </dev/null    # 670ms
```

My local RelWithDebInfo build is even worse at three seconds.

**Comment by tydeu** [comment]
Is this `lake serve` in particular or just Lake in general (e.g., `lake build` as well)?

**Comment by tydeu** [comment]
What confuses me here is that there is no significant runtime difference between the v3.2.1 bootstrap [CI](https://github.com/leanprover/lake/runs/7205626361) and the current bootstrap [CI]( https://github.com/leanprover/lake/runs/7289622830). I would think such a performance degradation would effect the build as well.

**Comment by tydeu** [comment]
This issue is rather stale and some major performance improvements have landed ([lean4#2444](https://github.com/leanprover/lean4/pull/2444) and [lean4#2480](https://github.com/leanprover/lean4/pull/2480)), so I am closing this issue.
